File: Phidget Code/src/Models/In/MatchPlanning.java
package Models.In;

import java.util.ArrayList;
import java.util.Date;

/**
 * Created by bri_e on 07-04-17.
 */
public class MatchPlanning {

    private ArrayList<Match> planned;

    public MatchPlanning(){
        planned = new ArrayList<>();
    }

    public void addMatchToList(Date start, Date end){
        deleteOldMatches();
        planned.add(new Match(start,end));
    }

    public boolean areWeDuringAMatch(){
        deleteOldMatches();
        ArrayList<Match> temporaryList = (ArrayList<Match>) planned.clone();
        temporaryList.removeIf(s -> !s.areWeDuringMatch());
        if (temporaryList.isEmpty()){
            return false;
        } else {
            //Normaly the DB must contain only one match at one hour.
            return true;
        }
    }

    public ArrayList<Match> matchArrayList(){
        deleteOldMatches();
        return planned;
    }


    private void deleteOldMatches(){
        planned.removeIf(s -> s.isMatchFinished());
    }
}

File: PhidgetScala/src/Models/In/MatchPlanning.scala
package Models.In

import java.util
import java.util.{ArrayList, Date}

/**
  * Created by bri_e on 20-04-17.
  * Planning of every planned match. Contains an array of matches that are planned in th stadium
  */
class MatchPlanning() {

  // array of matches that are planned in th stadium
  var planned: util.ArrayList[Match] = new util.ArrayList[Match]

  /**
    * add a match to the planning
    * @param start start date of the new match
    * @param end end date of the new match
    */
  def addMatchToList(start: Date, end: Date): Unit = {
    deleteOldMatches()
    println("Filtered list of matches...")
    planned.add(new Match(start, end))

  }

  /**
    * @return if we are during a match
    */
  def areWeDuringAMatch: Boolean = {
    deleteOldMatches()
    val temporaryList: util.ArrayList[Match] = planned.clone.asInstanceOf[util.ArrayList[Match]]
    temporaryList.removeIf((s: Match) => !s.areWeDuringMatch)
    if (temporaryList.isEmpty)
      false
    else { //Normaly the DB must contain only one match at one hour.
      true
    }
  }

  /**
    * @return the actual list filtered from oldmatches
    */
  def matchArrayList: util.ArrayList[Match] = {
    deleteOldMatches()
    planned
  }

  /**
    * filtering old matches
    */
  private def deleteOldMatches(): Unit = {
    planned.removeIf((s: Match) => s.isMatchFinished)
  }
}
